rfd/0203-database-healthchecks.md

```diff
@@ -0,0 +1,607 @@
+| Teleport                       | 10s      | 5s                   |                    2 |                    1 |
+
+As a brief summary of the Teleport health checker behavior:
+
+1. it does the first check immediately
+2. the first heatlh check will always transition to `healthy` or `unhealthy` regardless of configured thresholds
+3. it repeats the check after interval time
+4. each check blocks until pass or fail
+
+For Teleport the minimum time initialized->healthy or initialized->unhealthy is roughly 0s.
+
+The maximum time is bounded by the timeout: 5s.
+
+Adding the 5s heartbeat polling period to the max times, we get a max time to broadcast the status change: 10s.
+
+Thus, broadcasting the health status for a new or updated resource will take between 0s and 10s.
+
+This is fast enough for web UI enrollment interactions that might wait for the resource network status to be reported.
+
+The default healthy threshold is set to a higher value than the unhealthy threshold to help avoid false positive `healthy` status when network connectivity is unreliable.
+It makes sense to bias the health status to `unhealthy` when the network is unreliable anyway.
+
+[^2]: https://docs.aws.amazon.com/elasticloadbalancing/latest/network/target-group-health-checks.html
+[^3]: https://learn.microsoft.com/en-us/azure/load-balancer/load-balancer-custom-probe-overview
+[^4]: Azure timeouts always match the interval, they cannot be configured independently.
+[^5]: https://docs.docker.com/reference/dockerfile/#healthcheck
+[^6]: https://cloud.google.com/load-balancing/docs/health-check-concepts
+[^7]: https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes
+
+#### Config example
+
+    $ tctl get health_check_config/default
+    kind: health_check_config
+    version: v3
+    metadata:
+      name: "default"
+      labels:
+        teleport.internal/resource-type: preset
+    spec:
+      enabled: true
+      match:
+        db_labels:
+          - name: '*'
+            values:
+            - '*'
+      interval: 10s
+      timeout: 5s
+      healthy_threshold: 2
+      unhealthy_threshold: 1
+
+#### Terraform config example
+
+    resource "teleport_health_check_config" "example" {
+      version = "v3"
+      metadata = {
+        name = "example"
+      }
+      spec = {
+        enabled = true
+        match = {
+          db_labels = [{
+            name = "env"
+            values = ["dev"]
+          }]
+          env = "dev"
+        }
+        healthy_threshold   = 1
+        unhealthy_threshold = 1
+        interval            = "30s"
+        timeout             = "5s"
+      }
+    }
+
+### DB agent behavior
+
+DB agents will cache `health_check_config` resources.
+
+When a DB agent begins proxying a database, the agent will get all `health_check_config` resources and determine which, if any, will apply to the database.
```

Comment:
could you document how will agent determine which hc config it will use when there are multiple configs match the dabase?